bindings/matrix-sdk-ffi/src/room_info.rs

```diff
@@ -57,6 +64,33 @@ impl RoomInfo {
             joined_members_count: room.joined_members_count(),
             highlight_count: unread_notification_counts.highlight_count,
             notification_count: unread_notification_counts.notification_count,
+            notification_mode: notification_mode(room).await,
         })
     }
 }
+
+/// Get the notification mode for a given `Room`
+async fn notification_mode(room: &matrix_sdk::Room) -> Option<RoomNotificationMode> {
+    if !matches!(room.state(), RoomState::Joined) {
+        return None;
+    }
+    let notification_settings = room.client().notification_settings().await;
+
+    // Get the user-defined mode if available
+    let sdk_mode =
```

Comment:
It's incorrect to have logic like this in the FFI layer, this should go elsewhere in the SDK base crate.

Reply:
@bnjbvr, I've moved the logic into the SDK, let me know if that works for you.